File: examples/quickstart-firefox/src/test/java/org/fluentlenium/examples/test/DuckDuckGoFirefoxTest.java
package org.fluentlenium.examples.test;

import org.fluentlenium.core.annotation.Page;
import org.fluentlenium.examples.pages.DuckDuckMainPage;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class DuckDuckGoFirefoxTest extends AbstractFirefoxTest {

    @Page
    private DuckDuckMainPage duckDuckMainPage;

    @Override
    public WebDriver newWebDriver() {
        return new FirefoxDriver();
    }

    @Test
    void titleOfDuckDuckGoShouldContainSearchQueryName() {
        String searchPhrase = "searchPhrase";

        goTo(duckDuckMainPage)
                .typeSearchPhraseIn(searchPhrase)
                .submitSearchForm()
                .assertIsPhrasePresentInTheResults(searchPhrase);
    }

}

File: examples/quickstart-firefox/src/test/java/io/fluentlenium/examples/test/basic/DuckDuckGoFirefoxTest.kt
package io.fluentlenium.examples.test.basic

import io.fluentlenium.core.annotation.Page
import io.fluentlenium.examples.pages.basic.DuckDuckMainPage
import io.fluentlenium.examples.test.AbstractFirefoxTest
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable
import org.openqa.selenium.WebDriver
import org.openqa.selenium.firefox.FirefoxDriver

@DisabledIfEnvironmentVariable(named = "CI", matches = "true")
class DuckDuckGoFirefoxTest : AbstractFirefoxTest() {

    @Page
    private lateinit var onDuckDuckGoMainPage: DuckDuckMainPage

    override fun newWebDriver(): WebDriver {
        return FirefoxDriver()
    }

    @Test
    fun titleOfDuckDuckGoShouldContainSearchQueryName() {
        val searchPhrase = "FluentLenium"
        goTo(onDuckDuckGoMainPage)

        onDuckDuckGoMainPage.perform {
            typeSearchPhraseIn(searchPhrase)
            submitSearchForm()
            assertIsPhrasePresentInTheResults(searchPhrase)
        }
    }
}